onnx/version_converter/IntraDomainConverter.h

```diff
@@ -0,0 +1,205 @@
+
+namespace ONNX_NAMESPACE { namespace version_conversion {
+
+class IntraDomainVersionConverter : BaseVersionConverter {
+  private:
+    bool DEBUG = false;
+
+  public:
+    IntraDomainVersionConverter() = default;
+
+    const Adapter& adapter_lookup(const Node* op,
+        const OpSetID& initial_version,
+        const OpSetID& target_version,
+        const std::unordered_map<const Node*, const OpSchema*>& current_opschemas) const {
+      const std::string& op_name = op->name();
+      const std::string& initial = initial_version.toString();
+      const std::string& target = target_version.toString();
+      // Find appropriate adapter in adapters map for provided initial and target versions
+      // TODO: Consider abstracting elements of this that are specific to
+      // DefaultConverter to separate methods here and maintain the procedure in Base Converter
+      const auto& op_adapters = adapters.find(op_name);
+      if (op_adapters != adapters.end()) {
+        // If we're adapting downwards, we just want to find the one downwards
+        // adapter implemented for initial_version. If we're adapting upwards, we
+        // want to actually use the SinceVersion value for the given op.
+        if (target_version.version() < initial_version.version()) {
+          // Downwards adapter
+          if (op_adapters->second.find(initial) != op_adapters->second.end()) {
+            // Either an upwards or a downwards adapter exists
+            // Check if downwards adapter exists (only one should)
+            const auto& target_map = op_adapters->second.at(initial);
+            for (auto it = target_map.begin(); it != target_map.end(); ++it) {
+              int64_t new_target = OpSetID::fromString(it->first).version();
+              if (new_target <= target_version.version()) {
+                // Adapter found
+                return *(it->second);
+              }
+            }
+            // If loop terminates, no downwards adapter was found
+            // TODO: Instead return OpAlreadyAtOldestVersion
+            throw "OpAlreadyAtOldestVersion";
+          } else {
+            // No adapters exist from initial_version
+            // TODO: Instead return NoAdapterForCurrentVersion
+            throw "NoAdapterForCurrentVersion";
+          }
+        } else {
+          // Upwards adapter
+          // Either adapt from SinceVersion or Incompatible Breaking Change
+          std::string since = target_version.domain() + std::to_string(
+              current_opschemas.at(op)->since_version());
+          const auto& op_adapters = adapters.at(op_name);
+          if (op_adapters.find(since) != op_adapters.end() &&
+            op_adapters.at(since).find(target) != op_adapters
+            .at(since).end()) {
+            return *(op_adapters.at(since).at(target));
+          } else {
+            // TODO: Instead return NoUpwardsAdapter
+            throw "NoUpwardsAdapter";
```

Comment:
follow the tensorerror in https://github.com/onnx/onnx/blob/master/onnx/common/assertions.h#L16.
Create a new type of error, or just use ONNX_ASSERTM

Please don't directly throw in .h file.